plugins/inventory/iocage.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024 Vladimir Botka <vbotka@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: iocage
+    short_description: iocage inventory source
+    version_added: 10.2.0
+    author:
+        - Vladimir Botka (@vbotka)
+    requirements:
+        - iocage >= 1.8
+    description:
+        - Get inventory hosts from the iocage jail manager.
+        - Uses a configuration file as an inventory source, it must end
+          in C(.iocage.yml) or C(.iocage.yaml).
+    extends_documentation_fragment:
+        - constructed
+        - inventory_cache
+    options:
+        plugin:
+            description:
+              - The name of this plugin, it should always be set to
+                V(community.general.iocage) for this plugin to recognize
+                it as its own.
+            required: true
+            choices: ['community.general.iocage']
+            type: str
+        host:
+            description: The ip/name of the C(iocage) host.
+            required: true
+            type: str
+        user:
+            description:
+              - C(iocage) user.
+                It is expected that the O(user) is able to connect to the
+                O(host) and execute the command C(iocage list).
+                This option is not required if O(host) is V(localhost).
+            type: str
+        get_properties:
+            description:
+              - Get jails' properties.
+                Creates dictionary C(iocage_properties) for each added host.
+            type: boolean
+            default: false
+        env:
+            description: O(user)'s environment on O(host).
+            type: list
+            elements: str
+            default: []
```

Comment:
I would argue that it makes more sense to have a `dict` here rather than `list`

Reply:
The argument would be that environment variables are, inherently, a mapping construct, they map var names to their respective values. That's the reason why `Popen`'s env argument expects a dict, not a list, same for `os.environ`. And I would assume that most, if not all, modules and plugins treat sets of environment vars as a dict rather than a list, so another potential argument would be consistency.

That being said, I'm not planning on pressing this point. If you still want to keep it as a list, feel free to.

Reply:
Yes. I want to keep it as a list. I oppose your arguments:

1) The parameter `env` in this plugin can't accept a variable.
2) We already clarified that `Popen` argument `env` can't be used here.
3) `shell> grep -r 'Popen' plugins/ | grep 'env'` shows nothing.
4) The consistency is a valid point. Limited by the plugin context.

In summary, I argue that the consistency here is not worth the dictionary complexity burden.